extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -12,41 +12,118 @@ class NodeImage extends Node {
+				$parsed = $this->getExternalParser()->parseRecursive( $parameterValue );
+
+				$items = array_merge( $this->getGalleryItems( $parsed ), $this->getTabberItems( $parsed ) );
+				for( $i = 0; $i < count( $items ); $i++ ) {
+					$this->data[] = $this->getImage(
+						$items[$i]['title'],
+						$items[$i]['caption'],
+						$items[$i]['caption']
+					);
+				}
+			} else {
+				$this->data[] = $this->getImage(
+					$parameterValue,
+					$this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} ),
+					$this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} )
+				);
 			}
+		}
+
+		return $this->data;
+	}
+
+	private function getGalleryItems( $html ) {
+		$items = array();
+
+		if ( preg_match( '#\sdata-model="([^"]+)"#', $html, $galleryOut ) ) {
+			$model = json_decode( htmlspecialchars_decode( $galleryOut[1] ), true );
+			$items = array_map( function( $modelItem ) {
+				return array(
+					'title' => $modelItem['title'],
+					'caption' => $modelItem[ 'caption' ]
+				);
+			}, $model );
+		}
 
-			$title = $this->getImageAsTitleObject( $imageData );
-			$file = $this->getFilefromTitle( $title );
-			if ( $title instanceof \Title ) {
-				$this->getExternalParser()->addImage( $title->getDBkey() );
+		if ( preg_match_all('#data-image-key="([^"]+)".*?\s<h2>(.*?)<\/h2>#is', $html, $galleryOut ) ) {
+			for( $i = 0; $i < count( $galleryOut[0] ); $i++ ) {
+				$items[] = array(
+					'title' => $galleryOut[1][$i],
+					'caption' => $galleryOut[2][$i]
+				);
 			}
-			$ref = null;
-			$alt = $this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} );
-			$caption = $this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} );
-
-			$this->data = [
-				'url' => $this->resolveImageUrl( $file ),
-				'name' => ( $title ) ? $title->getText() : '',
-				'key' => ( $title ) ? $title->getDBKey() : '',
-				'alt' => $alt,
-				'caption' => $caption
-			];
-
-			if ( $this->isVideo( $file ) ) {
-				$this->data = $this->videoDataDecorator( $this->data, $file );
+		}
+
+		return $items;
+	}
+
+	private function getTabberItems( $html ) {
+		$items = array();
+
+		if ( preg_match_all('/class="tabbertab" title="([^"]+)".*?\sdata-image-key="([^"]+)"/is', $parsed, $tabberOut) ) {
+			for( $i = 0; $i < count( $tabberOut[0] ); $i++ ) {
+				$items[] = array(
+					'title' => $tabberOut[2][$i],
+					'caption' => $tabberOut[1][$i]
+				);
 			}
 		}
 
-		return $this->data;
+		return $items;
+	}
+
+	/**
+	 * @desc Checks if parser preprocessed string containg Tabber or Gallery extension
+	 * @param string $str String to check
+	 * @return bool
+	 */
+	private function containsTabberOrGallery( $str ) {
+		// TODO: Consider more robust approach (UNIQ...QINU)
+		$strLower = strtolower( $str );
+		if ( strpos( $strLower, '-tabber-' ) !== false || strpos( $strLower, '-gallery-' ) !== false ) {
+			return true;
+		}
+		return false;
+	}
+
+	private function getImage( $parameterValue, $alt, $caption ) {
```

Comment:
maybe we can make this function look like: `getImage( $parameterValue, $caption, $alt = null )` and by this not to pass duplicated and redundant params here: https://github.com/Wikia/app/pull/8703/files#diff-a4400e3b90d497838c5bc1981a0b4da6R27 ?